src/library_webgl.js

```diff
@@ -1784,35 +1798,50 @@ var LibraryGL = {
       GL.timerQueriesEXT[id] = null;
     }
   },
+#endif
 
+#if !USE_WEBGL2 // WebGL2 alias defined in library_webgl2.js
   glIsQueryEXT__sig: 'ii',
   glIsQueryEXT: function(id) {
     var query = GL.timerQueriesEXT[id];
     if (!query) return 0;
     return GLctx.disjointTimerQueryExt['isQueryEXT'](query);
   },
+#endif
 
+#if !USE_WEBGL2 // WebGL2 alias defined in library_webgl2.js
   glBeginQueryEXT__sig: 'vii',
   glBeginQueryEXT: function(target, id) {
 #if GL_ASSERTIONS
     GL.validateGLObjectID(GL.timerQueriesEXT, id, 'glBeginQueryEXT', 'id');
 #endif
     GLctx.disjointTimerQueryExt['beginQueryEXT'](target, GL.timerQueriesEXT[id]);
   },
+#endif
 
+#if !USE_WEBGL2 // WebGL2 alias defined in library_webgl2.js
   glEndQueryEXT__sig: 'vi',
   glEndQueryEXT: function(target) {
     GLctx.disjointTimerQueryExt['endQueryEXT'](target);
   },
+#endif
 
+  // This one is either from EXT_disjoint_timer_query on WebGL 1 (taking a
+  // WebGLTimerQueryEXT) or from EXT_disjoint_timer_query_webgl2 (taking a
+  // WebGLQuery)
   glQueryCounterEXT__sig: 'vii',
   glQueryCounterEXT: function(id, target) {
 #if GL_ASSERTIONS
     GL.validateGLObjectID(GL.timerQueriesEXT, id, 'glQueryCounterEXT', 'id');
 #endif
+#if USE_WEBGL2
+    GLctx.disjointTimerQueryExt['queryCounterEXT'](GL.queries[id], target);
```

Comment:
Yes. Both extensions expose the same queryCounterEXT function.

Reply:
I think instead of using two arrays `GL.queries` and `GL.timerQueriesEXT`, let's just use one array to simplify the code, i.e. let's remove `GL.timerQueriesEXT` array altogether, and lump that in with `GL.queries`? I don't think there is a scenario where we should distinguish between these two.

Reply:
Done -- `GL.timerQueriesEXT` renamed to `GL.queries`.